File: clojure/fibonacci.clj
(defn fibonacci [n] 
    (if (< n 2) 1
    (loop [i 2 pre-pre 1 pre 1]
        (def current (+ pre-pre pre))
        (if (>= i n) current
            (recur (inc i) pre current))))
)
(def nn (read-string (nth  *command-line-args* 0)))
(println (map fibonacci (range 2 nn)))

File: clojure/fibonacci.rb
def fibonacci(n)
    return 1 if n < 2
    pre = 1
    val = 1
    (2..n).each { 
        tmp = val
        val = val + pre
        pre = tmp
    }
    return val
end
print (1..25).map {|i| fibonacci i}
